File: movie-ticket-system/movie-service/app/routers/movie_routes.py
from fastapi import APIRouter, Depends, Query
from app.schemas.movie_schema import MovieCreate, MovieUpdate
from app.services.movie_service import create_movie, get_all_movies, get_movie, update_movie, delete_movie
from app.middleware.auth import require_hall_owner_or_admin

router = APIRouter()


@router.post("/movies")
def create(movie: MovieCreate, user=Depends(require_hall_owner_or_admin)):
    movie_id = create_movie(movie.model_dump())
    return {"movie_id": movie_id}


@router.get("/movies")
def movies(hallId: str = Query(None, alias="hallId")):
    return get_all_movies(hall_id=hallId)


@router.get("/movies/{id}")
def movie(id: str):
    return get_movie(id)


@router.put("/movies/{id}")
def update(id: str, movie: MovieUpdate, user=Depends(require_hall_owner_or_admin)):
    data = {k: v for k, v in movie.model_dump().items() if v is not None}
    if data:
        update_movie(id, data)
    return {"message": "Movie updated"}


@router.delete("/movies/{id}")
def delete(id: str, user=Depends(require_hall_owner_or_admin)):
    delete_movie(id)
    return {"message": "Movie deleted"}

File: movie-ticket-system/movie-service/src/routes/movie.routes.js
const express = require("express");
const {
  createMovie,
  getAllMovies,
  getMovieById,
  updateMovieById,
  deleteMovieById,
} = require("../services/movie.service");
const { requireHallOwnerOrAdmin } = require("../middleware/auth.middleware");

const router = express.Router();

router.post("/", requireHallOwnerOrAdmin, async (req, res) => {
  try {
    const movieId = await createMovie(req.body || {});
    return res.json({ movie_id: movieId });
  } catch (err) {
    console.error("Error creating movie:", err);
    return res.status(500).json({ detail: "Internal server error" });
  }
});

router.get("/", async (req, res) => {
  try {
    const { hallId } = req.query;
    const movies = await getAllMovies(hallId);
    return res.json(movies);
  } catch (err) {
    console.error("Error fetching movies:", err);
    return res.status(500).json({ detail: "Internal server error" });
  }
});

router.get("/:id", async (req, res) => {
  try {
    const movie = await getMovieById(req.params.id);
    if (!movie) {
      return res.status(404).json({ detail: "Movie not found" });
    }
    return res.json(movie);
  } catch (err) {
    console.error("Error fetching movie:", err);
    return res.status(500).json({ detail: "Internal server error" });
  }
});

router.put("/:id", requireHallOwnerOrAdmin, async (req, res) => {
  try {
    const updated = await updateMovieById(req.params.id, req.body || {});
    if (!updated) {
      return res.status(404).json({ detail: "Movie not found" });
    }
    return res.json({ message: "Movie updated" });
  } catch (err) {
    console.error("Error updating movie:", err);
    return res.status(500).json({ detail: "Internal server error" });
  }
});

router.delete("/:id", requireHallOwnerOrAdmin, async (req, res) => {
  try {
    const deleted = await deleteMovieById(req.params.id);
    if (!deleted) {
      return res.status(404).json({ detail: "Movie not found" });
    }
    return res.json({ message: "Movie deleted" });
  } catch (err) {
    console.error("Error deleting movie:", err);
    return res.status(500).json({ detail: "Internal server error" });
  }
});

module.exports = router;

